Servlet 6.x (EE11) demos

gregw opened the pull request:
Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO

gregw commented:
@janbartel @sbordet @lachlan-roberts I will soon be deleting all the remaining code from `./jetty-ee11/jetty-ee11-demos/jetty-ee11-demo-embedded`.   If you have any HTTP2, websocket or other tests from there that you think we need to keep, then can you collaborate on this branch and add them to `jetty-demos/jetty-servlet6-demos` in some form or other.

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO
 + ambiguous URIs

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO
 + ambiguous URIs

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO
 + ambiguous URIs

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO
 + ambiguous URIs
 + moved EE11 embedded demos to servlet6

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO
 + ambiguous URIs
 + moved EE11 embedded demos to servlet6

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO
 + ambiguous URIs
 + moved EE11 embedded demos to servlet6

 Fixed CachingHttpContentFactory

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO
 + ambiguous URIs
 + moved EE11 embedded demos to servlet6

 Fixed CachingHttpContentFactory

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO
 + ambiguous URIs
 + moved EE11 embedded demos to servlet6

 Fixed CachingHttpContentFactory

gregw committed:
spotless

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO
 + ambiguous URIs
 + moved EE11 embedded demos to servlet6

 Fixed CachingHttpContentFactory

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO
 + ambiguous URIs
 + moved EE11 embedded demos to servlet6

 Fixed CachingHttpContentFactory

lorban commented on jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java:
```diff
@@ -354,23 +367,40 @@ public String getKey()
+                    buffer.retain();
+                    try
+                    {
+                        sink.write(true, BufferUtil.slice(buffer.getByteBuffer(), (int)offset, (int)length), Callback.from(buffer::release, callback));
```
Don't you want to also null out the buffer ref in the callback?

lorban committed:
#12681 atomically retain the buffer with a check that it still is in the cache, otherwise delegate to the wrapped content

Signed-off-by: Ludovic Orban <lorban@bitronix.be>

lorban committed:
#12681 remove invalid comment

Signed-off-by: Ludovic Orban <lorban@bitronix.be>

lorban committed:
#12681 add tests

Signed-off-by: Ludovic Orban <lorban@bitronix.be>

gregw committed:
Fix servlet 6.x demos

Remove disabled embedded examples
Added servlet 6.x examples for
 + durable dispatch paths
 + ByteBuffer IO
 + ambiguous URIs
 + moved EE11 embedded demos to servlet6

 Fixed CachingHttpContentFactory

gregw committed:
Merge branch 'jetty-12.1.x' into fix/jetty-12.1.x/serlvet6-demos

# Conflicts:
#	jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java

gregw committed:
Merge remote-tracking branch 'origin/fix/jetty-12.1.x/12681-cached-content-already-released' into fix/jetty-12.1.x/serlvet6-demos

# Conflicts:
#	jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java

gregw committed:
Merge remote-tracking branch 'origin/fix/jetty-12.1.x/12681-cached-content-already-released' into fix/jetty-12.1.x/serlvet6-demos

# Conflicts:
#	jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java

lorban commented:
I assume the `CachingHttpContentFactory` changes can go away once https://github.com/jetty/jetty.project/pull/12704 has been merged?

gregw committed:
rename

gregw committed:
Merge remote-tracking branch 'origin/jetty-12.1.x' into fix/jetty-12.1.x/serlvet6-demos

gregw committed:
fix module

gregw committed:
fix module

janbartel commented on jetty-home/pom.xml:
```diff
@@ -292,6 +292,16 @@
       <type>jar</type>
       <optional>true</optional>
     </dependency>
+    <!-- Demo Servlet 6 Apps -->
```
Where are the other new servlet6 webapps?

gregw committed:
updates from review
